server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts

```diff
@@ -514,19 +514,29 @@ export class SummaryWriter implements ISummaryWriter {
             (missingOps.concat(logTail)).sort((op1, op2) => op1.sequenceNumber - op2.sequenceNumber) :
             logTail;
 
-        // Check the missing operations in the fullLogTail
-        if (fullLogTail.length !== (to - from - 1)) {
+        // Check the missing operations in the fullLogTail. We would treat
+        // isLogtailEntriesMatch as true when the fullLogTail's length is extact same as the requested range.
+        // As well as the first SN of fullLogTail - 1 is equal to from,
+        // and the last SN of fullLogTail + 1 is equal to to.
+        // For example, from: 2, to: 5, fullLogTail with SN [3, 4] is the true scenario.
+        const isLogtailEntriesMatch = fullLogTail &&
+            fullLogTail.length > 0 &&
+            fullLogTail.length === (to - from - 1) &&
+            from === fullLogTail[0].sequenceNumber - 1 &&
+            to === fullLogTail[fullLogTail.length - 1].sequenceNumber + 1;
+        if (!isLogtailEntriesMatch) {
             const missingOpsSequenceNumbers: number[] = [];
-            const fullLogTailSequenceNumbers = fullLogTail.map((ms) => ms.sequenceNumber);
-            let j = 0;
+            const fullLogTailSequenceNumbersSet = new Set();
+            const fullLogTailSequenceNumbers = fullLogTail?.map((ms) => ms.sequenceNumber);
+            fullLogTailSequenceNumbers.forEach((op) => fullLogTailSequenceNumbersSet.add(op));
```

Comment:
Can we instead do:
```ts
const fullLogTailSequenceNumbersSet = new Set();
fullLogTail?.map((ms) => fullLogTailSequenceNumbersSet.add(ms.sequenceNumber));
```
This way, you would only iterate through `fullLogTail` once. In the original approach above, you are iterating through that array 2 times. Once with `.map()` and once in the `.forEach()`

Reply:
Also, what does `ms` stand for? This is a minor nitpick, but maybe we can use `op` instead

Reply:
Thanks for the suggestions. Done.